include/seqan3/search/fm_index/bi_fm_index_iterator.hpp

```diff
@@ -126,6 +126,12 @@ class bi_fm_index_iterator
     size_type rev_rb;
     //\}
 
+    //!\brief Alphabet size of the index without delimiters
+    underlying_rank_t<typename index_type::char_type> sigma;
```

Comment:
Mhm, in theory yes. But why would you build an index over the mask alphabet 🤔 
But we can do something more fail safe here.

Reply:
the rank type of char is also uint8_t, but it's size cannot be represented by uint8_t!

Reply:
But it has to, since we work with `int_vector<8>`. That's why we have to throw if we find 255 (single) or 254/255 (collections), because we can otherwise not fit the alphabet into `uint8_t`.

Reply:
Yes, but the alphabet_size of `char` is still 256 which doesn't fit into `uint8_t`!! You can choose to ignore/misuse the last one or two values, but that does not change the alphabet_size.

Reply:
Ah, yeah right.

I can just expose the sigma type via `using sdsl_sigma_type = typename sdsl_index_type::alphabet_type::sigma_type;` in the fm_index, and then use this (though it's always an uint16_t for the implemented alphabets.

Reply:
Or just use size_t?

Reply:
Would also work. I guess it's till cheap to copy?